src/Util/StaticInstanceOf.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Util;
+
+use InvalidArgumentException;
+use Webmozart\Assert\Assert;
+
+/**
+ * @see \Rector\Core\Tests\Util\StaticInstanceOfTest
+ */
+final class StaticInstanceOf
+{
+    /**
+     * @param class-string[] $array
+     */
+    public static function isOneOf(?object $object, array $array): bool
+    {
+        if ($object === null) {
+            return false;
+        }
+
+        try {
+            Assert::isAnyOf($object, $array);
```

Comment:
updated with Webmozart's Assert isAnyOf(), wrapped with try catch to return false.

Reply:
it seems the check is not works for object with property call:

```bash

  src/PhpParser/Node/Manipulator/IdentifierManipulator.php:93
 ------------------------------------------------------------------------
  - '#Access to an undefined property PhpParser\\Node\:\:\$name#'
 ------------------------------------------------------------------------
 ```
 
 while actually it already works. I will retry with `instanceof` check instead of previous `is_a`